nx.json

```diff
@@ -1,88 +1,162 @@
 {
   "namedInputs": {
-    "default": ["{projectRoot}/**/*", "sharedGlobals"],
-    "sharedGlobals": ["{workspaceRoot}/.github/workflows/ci=github.yml"],
+    "default": [
+      "{projectRoot}/**/*",
+      "!{projectRoot}/LICENSE",
+      "!{workspaceRoot}/codecov.yml",
+      "!{workspaceRoot}/*.md",
+      "!{workspaceRoot}/.whitesource",
+      "!{workspaceRoot}/commitlint.config.js",
+      "!{workspaceRoot}/renovate.json",
+      "!{workspaceRoot}/.editorconfig",
+      "!{workspaceRoot}/.czrc",
+      "!{workspaceRoot}/.github/pull_request_template.md",
+      "!{workspaceRoot}/.github/workflows/*",
+      "!{workspaceRoot}/.husky/*",
+      "!{workspaceRoot}/.changeset/*.md",
+      "!{workspaceRoot}/contributing_docs/*.md",
+      "!{workspaceRoot}/.verdaccio/*",
+      "sharedGlobals"
+    ],
+    "sharedGlobals": ["{workspaceRoot}/.github/workflows/ci.yml"],
     "production": [
       "default",
       "!{projectRoot}/**/?(*.)+(spec|test).[jt]s?(x)?(.snap)",
+      "!{projectRoot}/**/?(*.)+(test-d).[jt]s?(x)?(.snap)",
+      "!{projectRoot}/tsconfig.spec.json",
+      "!{projectRoot}/src/test-setup.[jt]s",
+      "!{projectRoot}/test-setup.[jt]s",
+      "!{projectRoot}/eslint.config.mjs",
+      "!{projectRoot}/*.md",
+      "!{projectRoot}/**/*.md",
+      "!{projectRoot}/LICENSE"
+    ],
+    "noMarkdown": ["!{projectRoot}/*.md", "!{projectRoot}/**/*.md"],
+    "noTests": [
+      "!{projectRoot}/**/?(*.)+(spec|test).[jt]s?(x)?(.snap)",
+      "!{projectRoot}/**/?(*.)+(test-d).[jt]s?(x)?(.snap)",
       "!{projectRoot}/tsconfig.spec.json",
       "!{projectRoot}/src/test-setup.[jt]s",
-      "!{projectRoot}/test-setup.[jt]s"
+      "!{projectRoot}/test-setup.[jt]s",
+      "!{projectRoot}/eslint.config.mjs"
     ],
-    "noMarkdown": ["!{projectRoot}/**/*.md"]
+    "noEslintConfig": ["!{projectRoot}/eslint.config.mjs"]
   },
   "targetDefaults": {
-    "typecheck": {
-      "dependsOn": ["build", "^build"]
+    "nx-release-publish": {
+      "options": {
+        "packageRoot": "packages/{projectName}"
+      }
     },
-    "docs": {
-      "dependsOn": ["build", "^build", "^docs"],
-      "cache": true,
-      "outputs": ["{projectRoot}/docs"]
+    "typecheck": {
+      "outputs": ["{projectRoot}/out-tsc"],
+      "dependsOn": ["^typecheck"]
     },
     "serve": {
-      "cache": false,
-      "dependsOn": ["build"]
+      "dependsOn": ["^build"]
     },
     "e2e": {
-      "dependsOn": ["^build"],
-      "inputs": ["noMarkdown", "^noMarkdown"],
+      "dependsOn": ["^e2e"],
+      "inputs": [
```

Comment:
here are the `namedInputs`, defined above, being used.

Here we say ( in English) "do not consider markdown inputs from the project (e2e project), the child projects (apps, packages that are being consumed), do not consider eslintConfig from the project or child projects"